ci-operator/step-registry/baremetal/lab/upi/conf/grub2/baremetal-lab-upi-conf-grub2-commands.sh

```diff
@@ -55,7 +55,7 @@ load_video
 menuentry 'Install ($flavor)' {
     set gfx_payload=keep
     insmod gzio
-    linux  /${CLUSTER_NAME}/vmlinuz_${arch} debug nosplash ip=${baremetal_iface}:dhcp $kargs coreos.live.rootfs_url=http://${INTERNAL_NET_IP}/${CLUSTER_NAME}/rootfs-${arch}.img ignition.config.url=http://${INTERNAL_NET_IP}/${CLUSTER_NAME}/$mac_postfix-console-hook.ign ignition.firstboot ignition.platform.id=metal
+    linux  /${CLUSTER_NAME}/vmlinuz_${arch} debug nosplash ip=${baremetal_iface}:dhcp $kargs coreos.live.rootfs_url=http://${INTERNAL_NET_IP}/${CLUSTER_NAME}/rootfs-${arch}.img ignition.firstboot ignition.platform.id=metal
```

Comment:
the console-hook is needed for other kind of installations. 

We can add a parameter for this step, `USE_CONSOLE_HOOK`, set to `"true"` by default, and, before line 49, add
```shell
[ "$USE_CONSOLE_HOOK" == "true" ] && kargs="${kargs} ignition.config.url=http://${INTERNAL_NET_IP}/${CLUSTER_NAME}/$mac_postfix-console-hook.ign
```

Then, you can set the `USE_CONSOLE_HOOK` to false in the agent-related scenarios.

Reply:
These changes were added just for testing purposes. Yes, I have a plan to add a flag for the console hook.